spec/controllers/password_resets_controller_spec.rb

```diff
@@ -0,0 +1,85 @@
+require 'spec_helper'
+
+describe PasswordResetsController do
+  describe "GET show" do
+    context "with valid token" do
+      
+      let!(:valid_user) { object_generator(:user, token: '123456789') }
+      before { get :show, id: '123456789' }
+      
+      it "renders show template" do
+        expect(response).to render_template :show
+      end
+      
+      it "sets @token" do
+        expect(assigns(:token)).to eq('123456789')
+      end
+    end 
+    
+    context "with invalid token" do
+      it "redirects to expired token page" do
+        get :show, id: '123456789'
+        expect(response).to redirect_to expired_token_path
+      end
+    end 
+  end
+  
+  describe "POST create" do
+    context "with valid token" do
+      
+      context "with password that fails validation" do
+        let!(:valid_user) { object_generator(:user, token: '123456789', password: 'old_password') }
+        before { post :create, token: '123456789', password: 'tiny' }
+        
+        it "renders the password_resets show template" do
+          expect(response).to render_template :show
+        end
+        
+        it "does not reset the password" do
+          expect(valid_user.password).to eq('old_password')
+        end
+        
+        it "shows error messages" do
+          expect(valid_user.errors).not_to be_nil
+        end
+        
+        it "assigns @user" do
+          expect(assigns(:user)).to eq(valid_user)
+        end
+        
+        it "assigns @token" do
+          expect(assigns(:token)).to eq('123456789')
+        end
+      end
+      
+      context "with password that passes validation" do
+        let!(:valid_user) { object_generator(:user, token: '123456789', password: 'old_password') }
+        before { post :create, token: '123456789', password: 'new_password' }
+      
+        it "redirects to the sign in page" do
```

Comment:
One of the ways how to write these 1 liner specs that have pretty much same message as the example code is `it { is_expected.to redirect_to sign_in_path }`.